src/scratch/ScratchRuntime.as

```diff
@@ -89,49 +145,342 @@ public class ScratchRuntime {
 		processEdgeTriggeredHats();
 		interp.stepThreads();
 		app.stagePane.commitPenStrokes();
+		
+		if (ready>=0) {
+			app.stagePane.countdown(count);
+		}
 	}
 
-//-------- recording test ---------
+//-------- recording video code ---------
 	public var recording:Boolean;
 	private var frames:Array = [];
-
+	private var sounds:Array = [];
+	private var recTimer:Timer;
+	private var baFlvEncoder:ByteArrayFlvEncoder;
+	private var position:int;
+	private var secs:Number;
+	private var alreadyDone:int;
+	
+	private var pSound:Boolean;
+	private var mSound:Boolean;
+	private var mouse:Boolean;
+	private var cursor:Boolean;
+	public var fullEditor:Boolean;
+	private var framerate:Number;
+	private var width:int;
+	private var height:int;
+	public var ready:int=-1;
```

Comment:
I changed most of the variable names to be more explicitly related to video recording, and also made ready use declared constants as values.